src_c/font.c

```diff
@@ -745,128 +718,55 @@ font_init(PyFontObject *self, PyObject *args, PyObject *kwds)
             goto error;
         }
         fontsize = (int)(fontsize * .6875);
-        if (fontsize <= 1)
-            fontsize = 1;
     }
 
-    /* SDL accepts UTF8 */
-    oencoded = pg_EncodeString(obj, "UTF8", NULL, NULL);
-    if (!oencoded || oencoded == Py_None) {
-        /* got a file object, or an error */
-        Py_XDECREF(oencoded);
-        oencoded = NULL;
-        PyErr_Clear();
-        goto fileobject;
-    }
-    filename = PyBytes_AS_STRING(oencoded);
-
-#if FONT_HAVE_RWOPS
-    /* Try opening the path through RWops first */
-    if (filename) {
-        rw = SDL_RWFromFile(filename, "rb");
-        if (rw != NULL) {
-            Py_BEGIN_ALLOW_THREADS;
-            font = TTF_OpenFontIndexRW(rw, 1, fontsize, 0);
-            Py_END_ALLOW_THREADS;
-        }
-        else {
-            /*
-            PyErr_Format(PyExc_IOError,
-                                 "unable to read font file '%.1024s'",
-                                 filename);
-            goto error;
-            */
+    rw = pgRWops_FromObject(obj);
 
-            /* silently ignore this failure. We will try opening the path
-               with fopen (pg_open_obj) again later.
-               RWops can open assets bundled with P4A on Android, but not
-               font_resource() paths */
-        }
-        if (font != NULL)
-            goto success;
-    }
-#endif
-
-    if (font == NULL) {
-        /*check if it is a valid file, else SDL_ttf segfaults*/
-        test = pg_open_obj(obj, "rb");
-        if (test == NULL) {
-            if (strcmp(filename, font_defaultname) == 0) {
-                PyObject *tmp;
-                PyErr_Clear();
-                tmp = font_resource(font_defaultname);
-                if (tmp == NULL) {
-                    if (!PyErr_Occurred()) {
-                        PyErr_Format(PyExc_IOError,
-                                     "unable to read font file '%.1024s'",
-                                     filename);
-                    }
-                    goto error;
-                }
-                Py_DECREF(obj);
-                obj = tmp;
-                filename = PyBytes_AS_STRING(obj);
-                test = pg_open_obj(obj, "rb");
-            }
-            if (test == NULL) {
-                if (!PyErr_Occurred()) {
-                    PyErr_Format(PyExc_IOError,
-                                 "unable to read font file '%.1024s'",
-                                 filename);
+    if (rw == NULL && PyUnicode_Check(obj)) {
+        PyObject *font_defaultname_string =
+            PyUnicode_FromString(font_defaultname);
+        if (!PyUnicode_Compare(font_defaultname_string, obj)) {
```

Comment:
perhaps it's better to use [PyUnicode_CompareWithASCIIString](https://docs.python.org/3/c-api/unicode.html#c.PyUnicode_CompareWithASCIIString) here, rather than converting to python string and comparing with `PyUnicode_Compare`. Not only is this more elegant code wise, it also has the advantage of not erroring, whereas the old code here can error, and that is not handled correctly